bugsnag-android-core/src/main/java/com/bugsnag/android/Error.java

```diff
@@ -108,6 +112,15 @@ public void toStream(@NonNull JsonStream writer) throws IOException {
             writer.name("threads").value(threadState);
         }
 
+        if (!failureReasons.isEmpty()) {
+            writer.name("failureReason").beginArray();
```

Comment:
I think that it may be better to name this something like `reportFailureDetails` or `notifierFailureDetails` so that it:
- Is clearer that it is an array (as it is pluralized)
- Is more generic than `failureReason` (as we may want to include diagnostic details that aren't the reason for failure)
- Is a bit clearer that it's not to do with the error that is being reported but is regarding the generation of the report itself

Reply:
Can do. I'll be raising this in another PR and will rename the fields/JSON `notifierFailureDetails`.